util/encoding/key_encoding.go

```diff
@@ -410,44 +448,102 @@ func floatMandE(f float64) (int, []byte) {
 	if f < 0 {
 		f = -f
 	}
-	i := 0
-	if f >= 1 {
-		for t := f; t >= 1; t /= 100 {
-			i++
-		}
+
+	// Use strconv.FormatFloat to handle the intricacies of determining how much
+	// precision is necessary to precisely represent f. The 'e' format is
+	// d.dddde±dd.
+	b := strconv.AppendFloat(nil, f, 'e', -1, 64)
+	if len(b) < 5 {
+		// The formatted float must be at least 5 bytes or something unexpected has
+		// occurred.
+		panic(fmt.Errorf("malformed float: %v -> %s", f, b))
+	}
+
+	// Parse the exponent.
```

Comment:
Interesting. I was mislead that the output values take the form "+02". Looks like AppendFloat generates either a two digit or three digit exponent. I'll fix.